### PR TITLE
[go/uk-ua] Add ukrainian translation for Go language

### PR DESCRIPTION
**Pull request by AstiaSun**
- [x] I solemnly swear that this is all original content of which I am the original author
- [x] Pull request title is prepended with `[language/lang-code]`
- [x] Pull request touches only one file (or a set of logically related files with similar changes made)
- [x] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)
- [x] If you've changed any part of the YAML Frontmatter, make sure it is formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - [x] Yes, I have double-checked quotes and field names!

### CODE REVIEW

File: uk-ua/go-ua.html.markdown [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+// Декоратори звична річ для багатьох мов програмування. В Go їх можна реалізувати
+// за допомогою літералів функцій, що приймають аргументи.
+func sentenceFactory(mystring string) func(before, after string) string {
+	return func(before, after string) string {
+		return fmt.Sprintf("%s %s %s", before, mystring, after) // новий рядок
+	}
+}
+
+func learnDefer() (ok bool) {
+	// Відкладене твердження змушує функцію посилатись по список. Список
```
Comment (Menelion) [inline uk-ua/go-ua.html.markdown]: *A defer statement pushes a function call onto a list.* — Твердження defer додає виклик функції до списку... якось так?

File: uk-ua/go-ua.html.markdown [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+
+// Оголошує pair як структуру з двома полями, цілими числами x та y.
+type pair struct {
+	x, y int
+}
+
+// Оголошує метод для типу pair. pair тепер реалізує Stringer, оскільки pair оголосив
+// всі методи в цьому інтерфейсі.
+func (p pair) String() string { // p тепер називається "приймачем"
+	// Sprintf - ще ожна функція з пакету fmt.
```
Comment (Menelion) [inline uk-ua/go-ua.html.markdown]: одна

File: uk-ua/go-ua.html.markdown [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+	// Крапка використовується, щоб звернутись до полів об'єкту p.
+	return fmt.Sprintf("(%d, %d)", p.x, p.y)
+}
+
+func learnInterfaces() {
+	// Синтаксис з використанням фігурних дужок називається "літералом стуктури".
```
Comment (Menelion) [inline uk-ua/go-ua.html.markdown]: структури

File: uk-ua/go-ua.html.markdown [hunk 4]
```diff
@@ -0,0 +1,450 @@
...
+	// Він застосовується до ініціалізованої структури. Оператор := оголошує
+	// на ініціалізовує p цією структурою.
```
Comment (Menelion) [inline uk-ua/go-ua.html.markdown]: та ініціалізовує?

File: uk-ua/go-ua.html.markdown [hunk 5]
```diff
@@ -0,0 +1,450 @@
...
+	// більше про проблему.
+	if _, err := strconv.Atoi("non-int"); err != nil { // _ ігнорує значення
+		// виводить помилку 'strconv.ParseInt: parsing "non-int": invalid syntax'
+		fmt.Println(err)
+	}
+	// Ми розглянемо інтерфейси дещо пізніше. А поки, розглянемо багатопоточність.
+	learnConcurrency()
+}
+
+// Канал с - це потокозохіщений об'єкт для спілкування між потоками.
```
Comment (Menelion) [inline uk-ua/go-ua.html.markdown]: Потокозахищений

File: uk-ua/go-ua.html.markdown [hunk 6]
```diff
@@ -0,0 +1,450 @@
...
+
+// inc виконує збільшення значення на 1. Ми використаємо його, щоб збільшувати
+// числа рівночасно.
+func learnConcurrency() {
+	// вже знайома функція make, яка раніше використовувалась для виділення пам'яті, 
+	// тут використовується для створення каналу. Make виділяє пам'ять та ініціалізує
+	// зрізи, асоційовані масиви та канали. Новостворений канал буде передавати 
+	// цілочисельні значення.
+	c := make(chan int)
+	// Запустити три одночасні ґорутини. Числа будуть збільшуватись рівночасно, імовіно
```
Comment (Menelion) [inline uk-ua/go-ua.html.markdown]: імовірно; 👍 за Ґ у ґорутинах :)

File: uk-ua/go-ua.html.markdown [hunk 7]
```diff
@@ -0,0 +1,450 @@
...
+	// Читаємо три результати з каналу та друкуємо їх.
+	// Порядок результатів - невідомий!
+	fmt.Println(<-c, <-c, <-c) // якщо канал знаходиться справа від оператора <-, 
+								// він виконує функцію "приймача".
+
+	cs := make(chan string)       // Ще один канал, який примає рядки.
+	ccs := make(chan chan string) // Канал каналів рядків.
+	go func() { c <- 84 }()       // Запустимо нову ґорутину, щоб надіслати значення в канал с.
+	go func() { cs <- "wordy" }() // Надсилаємо "wordy" в канал cs.
+	// Ключове слово select має подібний до синтаксис до switch, проте кожен кейс
```
Comment (Menelion) [inline uk-ua/go-ua.html.markdown]: синтаксис, подібний до switch?

File: uk-ua/go-ua.html.markdown [hunk 8]
```diff
@@ -0,0 +1,450 @@
...
+		fmt.Printf("it's a %T", i)
+	case <-cs: // або значення може бути проігнороване.
+		fmt.Println("it's a string")
+	case <-ccs: // Пустий канал, не готовий комунікувати.
+		fmt.Println("Не відбудеться.")
+	}
+	// На цьому етапі, значення було прочитане або з с або з cs. Одна з двох
+	// ґорутин завершилась, але інша все ще заблокована.
+
+	learnWebProgramming() // Go вмає й веб. Так, ти хочеш зробити це.
```
Comment (Menelion) [inline uk-ua/go-ua.html.markdown]: вміє?)

File: uk-ua/go-ua.html.markdown [hunk 9]
```diff
@@ -0,0 +1,450 @@
...
+}
+
+// Лиш одна функція з пакету http запускає веб сервер.
+func learnWebProgramming() {
+	
+	// перший аргумент ListenAndServe - це TCP адрес, який сервер буде слухати.
```
Comment (Menelion) [inline uk-ua/go-ua.html.markdown]: адреса, яку

File: uk-ua/go-ua.html.markdown [hunk 10]
```diff
@@ -0,0 +1,450 @@
...
+```
+
+## Подальше вивчення
+
+Основним джерелом всієї інформації про Go залишається [офіційна веб-сторінка](http://golang.org/). Там можна знайти уроки, інтерактивно пограти та багато про що почитати.
+Окрім туру, у [документації](https://golang.org/doc/) міститься інформація як писати чистий та ефективний код на Go, документація пакетів та окремих команд, а також історія релізів.
+
+Надзвичайно рекомендується ознайомитись із визначенням мови. Вона легко читається та на диво коротка (в порівнянні з іншими сучасними мовами).
+
+Можна погратись з кодом вище на [Go playground](https://play.golang.org/p/tnWMjr16Mm). Спробуй змінити його та запустити із свого браузера. Поміть, що можна використовувати [https://play.golang.org](https://play.golang.org) як [REPL](https://en.wikipedia.org/wiki/Read-eval-print_loop) до тестів та коду в твоєму браузері, без встановлення Go.
```
Comment (Menelion) [inline uk-ua/go-ua.html.markdown]: Є україномовна стаття: https://uk.wikipedia.org/wiki/REPL

File: uk-ua/go-ua.html.markdown [hunk 11]
```diff
@@ -0,0 +1,450 @@
...
+
+В списку для прочитання новичкам в Go - [вихідний код стандартної бібліотеки](http://golang.org/src/pkg/). Код всеосяжно задукоментований, тому є найкращим прикладом з боку зручного для прочитання та швидкості розуміння коду на цій мові програмування. Приведений стиль та ідіоми Go.
```
Comment (Menelion) [inline uk-ua/go-ua.html.markdown]: новачкам; закоментований

**Comment by AstiaSun** [comment]
@Menelion Thanks a lot for finding typos! I have updated the text

**Comment by Menelion** [comment]
@AstiaSun Again, not all of them are fixed, as far as I see.

**Comment by AstiaSun** [comment]
@Menelion Seems like I have missed one by mistake
Looks like I have resolved all comments.

**Comment by Menelion** [comment]
@AstiaSun Thanks so much, merged! :)

**Pull request merged**